libcudacxx/.upstream-tests/utils/libcudacxx/test/config.py

```diff
@@ -423,7 +423,7 @@ def configure_no_execute(self):
     def configure_ccache(self):
         use_ccache_default = os.environ.get('CMAKE_CUDA_COMPILER_LAUNCHER') is not None
         use_ccache = self.get_lit_bool('use_ccache', use_ccache_default)
-        if use_ccache:
+        if use_ccache and not self.cxx.is_nvrtc:
```

Comment:
Why is this needed? Won't sccache just do nothing when using nvrtc?

Reply:
sccache will hard error about an unsupported compiler

Reply:
We have a nvrtc.sh which does some dark magic

Reply:
@jrhemstad exactly. We stitch together a source file that does that and includes the source of a given test, and we build and run that as the test executable with lit.